### PR TITLE
rename fluent bit restarts total metric

### PR DESCRIPTION
**Pull request by dennis-ge**
**Description**

Changes proposed in this pull request:

- The metric should not be equal to the number of total fluent bit restarts but should show the total number the telemetry operator triggered restarting fluent bit.
- Therefore the metric is renamed to `telemetry_fluentbit_triggered_restarts_total`

**Related issue(s)**
Fixes https://github.com/kyma-project/kyma/issues/15063

### CODE REVIEW

File: components/telemetry-operator/main.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func main() {
 	}
 
 	restartsTotal := prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "telemetry_fluentbit_restarts_total",
+		Name: "telemetry_fluentbit_triggered_restarts_total",
```
Comment (chrkl) [inline components/telemetry-operator/main.go:151]: There should also be a ServerMonitor that might need a change.

Comment (dennis-ge) [reply components/telemetry-operator/main.go:151]: Fixed :)

**Review by chrkl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**